src/goto-harness/goto_harness_parse_options.cpp

```diff
@@ -145,15 +146,23 @@ int goto_harness_parse_optionst::doit()
   harness_generator->generate(
     goto_model, got_harness_config.harness_function_name);
 
-  filter_goto_model(goto_model, goto_model_without_harness_symbols);
-  auto harness_out = std::ofstream{got_harness_config.out_file};
-  dump_c(
-    goto_model.goto_functions,
-    true,
-    true,
-    false,
-    namespacet{goto_model.get_symbol_table()},
-    harness_out);
+  if(!has_suffix(got_harness_config.out_file, ".c"))
```

Comment:
Probably just my personal preference, but I'd switch the condition and branches, so thats is more of the form:
```
if(has_suffix(".c"))
{
/* stuff to dump-c */
}
else
{
/* Dump goto-binary */
}
```
As to me, that reads more 'natural' rather than having the semi-invisible negation operator in the condition :-)

Reply:
Good call, thanks, fixed!